Program.cs

```diff
@@ -1,10 +1,101 @@
-﻿namespace Training
-{
-    internal class Program
-    {
-        static void Main(string[] args)
-        {
-            Console.WriteLine("Hello, World!");
-        }
-    }
+﻿// ---------------------------------------------------------------------------------------
+// Training ~ A training program for new joinees at Metamation, Batch- July 2023.
+// Copyright (c) Metamation India.                                                
+// ---------------------------------------------------------------------------------------
+// Program.cs                                                                     
+// Program to print the given numbers in words or roman numerals.
+// ---------------------------------------------------------------------------------------
+
+namespace Training {
+   #region class Program --------------------------------------------------------------------------
+   /// <summary>Number to wrods and roman numerals</summary>
+   internal class Program {
+      #region Method ------------------------------------------------
+      /// <summary>This method asks the user whether to display the num in words 
+      /// or roman numbers and displays accordingly</summary>
+      static void Main () {
+         Console.WriteLine ("Enter a positive number: \n" +
+                   "(Condition: Limit to convert to words is 10 crores. Limit to convert to roman numerals is 3999.)");
+         if (!int.TryParse (Console.ReadLine (), out int num)) {
+            Console.WriteLine ("Invalid Input");
+            return;
+         }
+         if (num < 0) {
+            Console.WriteLine ("Enter positive number.");
+            return;
+         }
+         Console.Write ("Type (W) to convert to words or (R) to roman: ");
+         string choice = Console.ReadLine ().ToLower ();
+
+         if (choice == "w") Console.WriteLine (num is > 100000000 or < 0
+                            ? "Enter value within the limit."
+                            : num == 0 ? "Zero" : GetNumInWords (num));
+         else if (choice == "r") Console.WriteLine (num is 3999
+                                 ? "Enter value within the limit."
+                                 : num <= 0 ? "No equivalent roman number" : GetRomanNum (num));
+         else Console.WriteLine ("Invlaid Input.");
+      }
+
+      /// <summary>This method converts the number to words</summary>
+      /// <param name="num">The number to be converted</param>
+      /// <returns>Returns the number in words as string</returns>
+      static string GetNumInWords (int num) {
+         string numInWrds = "";
+         Dictionary<int, int> divisor = new () {
+            [1] = 1, [2] = 1, [3] = 100, [4] = 1000, [5] = 1000,
+            [6] = 100000, [7] = 100000, [8] = 10000000, [9] = 10000000
+         };
+         while (num > 0) {
+            (string words, num) = GetWords (num, divisor[num.ToString ().Length]);

```

Comment:
![image](https://github.com/RaaneshKumar/Training/assets/75829713/2c11001c-c2dc-4580-a9bf-59900d2b1bb4)
Look for intellisense recommendations